install-rust-toolchain.sh

```diff
@@ -643,6 +646,18 @@ if [[ -n "${EXPORT_FILE:-}" ]]; then
     else
         echo export PATH=\"${IDF_TOOL_GCC_PATH}:\$PATH\" >>"${EXPORT_FILE}"
     fi
+
+    if [[ -z "${ESP_IDF_VERSION}" ]]; then
+        if [[ "${BUILD_TARGET}" =~ "esp32s3" ]]; then
+            echo "export IDF_SYSROOT_ESP32S3=\"$IDF_SYSROOT_ESP32S3\"" >>"${EXPORT_FILE}"
+        fi
+        if [[ "${BUILD_TARGET}" =~ "esp32s2" ]]; then
+            echo "export IDF_SYSROOT_ESP32S2=\"$IDF_SYSROOT_ESP32S2\"" >>"${EXPORT_FILE}"
+        fi
+        if [[ "${BUILD_TARGET}" =~ esp32[,|\ ] || "${BUILD_TARGET}" =~ esp32$ ]]; then
+            echo "export IDF_SYSROOT_ESP32=\"$IDF_SYSROOT_ESP32\"" >>"${EXPORT_FILE}"
+        fi
```

Comment:
I had just copied these conditionals from `install_rust_xtensa_toolchain` but looking over it again now, they should just check whether the variables they're using are actually defined instead.